roles/openstack-setup/tasks/flavors.yml

```diff
@@ -1,14 +1,23 @@
 ---
-- name: is m1.tiny undersized?
-  shell: mysql -e "select root_gb from nova.instance_types where name='m1.tiny';" | grep 10
-  failed_when: False
-  changed_when: False
-  register: resize_tiny_flavor
-  run_once: true
-
-- name: bump root disk size on m1.tiny
-  shell: mysql -e "update nova.instance_types set root_gb=10 where name='m1.tiny';"
-  when: resize_tiny_flavor|failed
+- name: Add flavors
+  os_nova_flavor:
+    name: '{{ item.name }}'
+    ram: '{{ item.ram | int }}'
+    disk: '{{ item.disk | int }}'
+    vcpus: '{{ item.vcpus | int }}'
+    flavorid: '{{ item.flavorid | default(omit) }}'
+    rxtx_factor: '{{ item.rxtx_factor | default(omit) }}'
+    ephemeral: '{{ item.ephemeral | default(omit) }}'
+    is_public: '{{ item.is_public | default(omit) }}'
+    swap: '{{ item.swap | default(omit) }}'
+    state: present
+    auth:
+      auth_url: "{{ endpoints.auth_uri  }}"
+      project_name: admin
+      username: admin
+      password: "{{ secrets.admin_password  }}"
+  with_items: '{{ ironic.flavors | default([]) }}'
+  when: not ironic.enabled | default(False) | bool
```

Comment:
I'm not sure we need this. ironic.enabled is True in the example allinone which means this is skipped. More likely we want this to be when nova.enabled because if nova and ironic can't co-exist this should be caught there.